src/condition.cpp

```diff
@@ -553,11 +480,7 @@ void conditional_t<T>::set_npc_aim_rule( const JsonObject &jo )
 {
     const std::string &setting = jo.get_string( "npc_aim_rule" );
     condition = [setting]( const T & d ) {
-        auto rule = aim_rule_strs.find( setting );
-        if( rule != aim_rule_strs.end() ) {
-            return d.beta->rules.aim == rule->second;
-        }
-        return false;
+        return d.beta->has_ai_rule( "aim_rule", setting );
     };
```

Comment:
There's actually some subtle differences on the back end, but yeah, it could probably be consolidated.

Honestly, a lot of the talk_effect_fun_t and condition code is probably overdone and needs a redesign/refactor - I was writing way past the limits of my ability and cribbing from the existing dynamic_line interface without really understanding what I was doing.  But it works so I'll just hope someone smarter than me cleans it up at some point.